packages/geo/src/lib/layer/layer-legend/layer-legend.component.ts

```diff
@@ -60,4 +61,21 @@ export class LayerLegendComponent implements OnInit, OnDestroy {
     this.legendItems$.next(legendItems);
   }
 
+  computeItemTitle(layerLegend): Observable<string> {
+    const layerOptions = this.layer.dataSource.options as any;
+    if (layerOptions.type !== 'wms') {
+      return of(layerLegend.title);
+    /*} else if (!layerOptions.optionsFromCapabilities) {
+      return of(layerLegend.title);*/
+    } else {
+      const layers = layerOptions.params.layers.split(',');
+      const localLayerOptions = JSON.parse(JSON.stringify(layerOptions)); // to avoid to alter the original options.
```

Comment:
I don't think this is still needed.

Reply:
Yes it is. It's the only way to parse the correct layer without altering the original one.